include/language-factory.php

```diff
@@ -0,0 +1,451 @@
+	 *     @type string  $search_url      Optional. Home URL to use in search forms.
+	 *     @type string  $host            Optional. Host corresponding to this language.
+	 *     @type string  $flag_url        Optional. URL of the flag.
+	 *     @type string  $flag            Optional. HTML markup of the flag.
+	 *     @type string  $custom_flag_url Optional. URL of the custom flag if it exists.
+	 *     @type string  $custom_flag     Optional. HTML markup of the custom flag if it exists.
+	 *     @type int     $page_on_front   Optional. ID of the page on front in this language.
+	 *     @type int     $page_for_posts  Optional. ID of the page for posts in this language.
+	 * }
+	 *
+	 * @return PLL_Language|null A language object if given data pass sanitization, null otherwise.
+	 *
+	 * @phpstan-param array{
+	 *     term_props?: array{
+	 *         language: array{
+	 *             term_id: positive-int,
+	 *             term_taxonomy_id: positive-int,
+	 *             count: int<0, max>
+	 *         },
+	 *         term_language: array{
+	 *             term_id: positive-int,
+	 *             term_taxonomy_id: positive-int,
+	 *             count: int<0, max>
+	 *         }
+	 *     },
+	 *     term_id?: positive-int,
+	 *     term_taxonomy_id?: positive-int,
+	 *     count?: int<0, max>,
+	 *     tl_term_id?: positive-int,
+	 *     tl_term_taxonomy_id?: positive-int,
+	 *     tl_count?: int<0, max>,
+	 *     name: non-empty-string,
+	 *     slug: non-empty-string,
+	 *     locale: non-empty-string,
+	 *     w3c: non-empty-string,
+	 *     flag_code: non-empty-string,
+	 *     term_group: int,
+	 *     is_rtl: int<0, 1>,
+	 *     mo_id?: positive-int,
+	 *     facebook?: non-empty-string,
+	 *     home_url?: non-empty-string,
+	 *     search_url?: non-empty-string,
+	 *     host?: non-empty-string,
+	 *     flag_url?: non-empty-string,
+	 *     flag?: non-empty-string,
+	 *     custom_flag_url?: non-empty-string,
+	 *     custom_flag?: non-empty-string,
+	 *     page_on_front?:positive-int,
+	 *     page_for_posts?:positive-int
+	 * } $language_data
+	 */
+	public static function create( $language_data ) {
+		// Backward compatibility with deprecated properties.
+		$term_props = array(
+			'term_id'             => array( 'language', 'term_id' ),
+			'term_taxonomy_id'    => array( 'language', 'term_taxonomy_id' ),
+			'count'               => array( 'language', 'count' ),
+			'tl_term_id'          => array( 'term_language', 'term_id' ),
+			'tl_term_taxonomy_id' => array( 'term_language', 'term_taxonomy_id' ),
+			'tl_count'            => array( 'term_language', 'count' ),
+		);
+
+		foreach ( $term_props as $prop => $value ) {
+			if ( ! empty( $language_data[ $prop ] ) && empty( $language_data['term_props'][ $value[0] ][ $value[1] ] ) ) {
+				$language_data['term_props'][ $value[0] ][ $value[1] ] = $language_data[ $prop ];
+			}
+			unset( $language_data[ $prop ] );
+		}
```

Comment:
We need this for 2 reasons (and that's why properties are optional):
1. If someone "updates" the plugin manually by replacing the files, the transient will still contain the old properties and not the new ones.
2. We need to keep the old properties in the transient in case someone needs to roll back to an old version of the plugin.